arangod/Aql/OptimizerRules.cpp

```diff
@@ -2608,9 +2608,6 @@ void arangodb::aql::simplifyConditionsRule(Optimizer* opt,
       AstNode* simplified = plan->getAst()->traverseAndModify(root, visitor);
       if (simplified != root) {
         nn->expression()->replaceNode(simplified);
-      }
-      // cppcheck-suppress knownConditionTrueFalse
-      if (modifiedNode) {
```

Comment:
"if" always was false?
Could you explain for me why did it change to always true?

Reply:
@Dronplane explained.

Reply:
Looks like `modifiedNode` value is not used anymore (only written) and could be removed

Reply:
Fixed